### ISSUE TITLE
[graphql] Error running query: argument of type 'NoneType' is not iterable

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-uiserver#159]
**Describe the bug**

Trying a new query to retrieve the offline data, used in the Cylc UI (for the offline UI).

```
{
  "errors": [
    {
      "message": "argument of type 'NoneType' is not iterable",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "workflows"
      ]
    }
  ],
  "data": {
    "workflows": null
  }
}
```

**Release version(s) and/or repository branch(es) affected?**

`master` for UIS (01988bb66307e53a1bda66e9ead70c64e0cda9c9) and Flow (bd293959edcf6db96414d608ef43772299514e36)

**Steps to reproduce the bug**

Run the query in GraphiQL, with a workflow on hold.

**Expected behavior**

Valid data retrieved, or an error response with the explanation.

**Screenshots**

**Additional context**

Query

```
query ($workflowID: ID) {
  workflows (ids: [$workflowID]) {
    ...WorkflowData
    cyclePoints: familyProxies (ids: ["root"], ghosts: true) {
      ...CyclePointData
    }
    taskProxies (sort: { keys: ["name"], reverse: false }, ghosts: true) {
      ...TaskProxyData
      jobs(sort: { keys: ["submit_num"], reverse:true }) {
        ...JobData
      }
    }
    familyProxies (exids: ["root"], sort: { keys: ["name"] }, ghosts: true) {
      ...FamilyProxyData
    }
  }
}

fragment WorkflowData on Workflow {
  id
  name
  status
  owner
  host
  port
}

fragment CyclePointData on FamilyProxy {
  id
  cyclePoint
}

fragment FamilyProxyData on FamilyProxy {
  id
  name
  state
  cyclePoint
  firstParent {
    id
    name
    cyclePoint
    state
  }
}

fragment TaskProxyData on TaskProxy {
  id
  name
  state
  isHeld
  cyclePoint
  latestMessage
  firstParent {
    id
    name
    cyclePoint
    state
  }
  task {
    meanElapsedTime
    name
  }
}

fragment JobData on Job {
  id
  firstParent: taskProxy {
    id
  }
  batchSysName
  batchSysJobId
  host
  startedTime
  submittedTime
  finishedTime
  state
  submitNum
}
```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by kinow** [comment]
Huh, strange, I force-refreshed the browser, and copy-paste'd the query. Then edited the variables, and it seems to be working now? :confused:

**Comment by kinow** [comment]
Ah, the error can be reproduced when you have a typo in the variable name :laughing: 

```
query (abc: [$abc]) { ...}
```

And variables has

```
{
  ab: ''
}
```

**Comment by kinow** [comment]
Debugging it, it looks like middlewares are executed before the error occurs. I thought it would first validate that we have the right variables in the context :thinking:
